storage/innobase/handler/ha_innodb.cc

```diff
@@ -16091,6 +16091,8 @@ ha_innobase::external_lock(
 
 	reset_template();
 	switch (sql_command) {
+	case SQLCOM_CREATE_TABLE:
+		break;
```

Comment:
A safer alternative might be to modify the `default:` case so that we will apply and buffered bulk insert to *all* of `trx->mod_tables` (and not only to `m_prebuilt->table`) when we observe the next non-INSERT statement. However, if we do that, I think that we should double check what would happen on a `CREATE TABLE…SELECT` on a partitioned table. (I checked from the grammar that it should be supported.) If that would lead to trouble, then this `case SQLCOM_CREATE_TABLE:` can be preserved, with a comment that explains why we need it.

Reply:
The intention is to finish any previously buffered bulk insert operation at the start of anything else than a plain `INSERT` statement. This had better be done also on `INSERT…ON DUPLICATE KEY UPDATE` and on `REPLACE` statements, which I assume might be overlooked in the existing check.